### PR TITLE
Re-index ES form and case indices

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Follow on from https://github.com/dimagi/commcare-hq/pull/10601
- Add `owner_type` and `user_type` properties to case and form indexes
- Add `inserted_at` property to case and form indexes

@benrudolph @dimagi/scale-team

### CODE REVIEW

**Comment by NoahCarnahan** [comment]
(don't merge until reindex has be completed)

**Comment by NoahCarnahan** [comment]
It would be good to just get a quick confirmation of the `inserted_at` commit, then I'll start the cases reindex.

**Comment by czue** [comment]
lg2m. i think you'll have to update a couple tests in `test_pillowtop` and `test_elasticsearch`

File: corehq/pillows/xform.py [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import collections
 import copy
+import datetime
+import pytz
```
Comment (dimagimon) [inline corehq/pillows/xform.py]: F401 'pytz' imported but unused

File: corehq/pillows/case.py [hunk 2]
```diff
@@ -1,4 +1,7 @@
 import copy
+import datetime
+import pytz
```
Comment (dimagimon) [inline corehq/pillows/case.py]: F401 'pytz' imported but unused

File: corehq/pillows/xform.py [hunk 3]
```diff
@@ -97,7 +97,7 @@ def transform_xform_for_elasticsearch(doc_dict, include_props=True):
         except KeyError:
             user_id = None
         doc_ret['user_type'] = get_user_type(user_id)
-        doc_ret['inserted_at'] = datetime.datetime.now(pytz.utc).isoformat()
+        doc_ret['inserted_at'] = datetime.datetime.utcnow().isoformat()
```
Comment (NoahCarnahan) [inline corehq/pillows/xform.py]: @benrudolph what's the advantage of excluding the timezone?

Comment (benrudolph) [inline corehq/pillows/xform.py]: we store timezone naive dates and our ES config doesn't allow us to submit timezone aware dates which is why the reindex kept failing. see: https://github.com/dimagi/commcare-hq/blob/master/corehq/pillows/core.py#L3. there's no format for the `+00:00` timezone. odd that it works locally, but think that's a configuration thing

Comment (NoahCarnahan) [inline corehq/pillows/xform.py]: Ah, interesting.

**Comment by benrudolph** [comment]
hot damn! time to merge this bad boy. full steam ahead!! 🚂 🚂

**Comment by czue** [comment]
build failed, looks likely real
```
======================================================================
FAIL: testConvertAndRestoreReportXFormDicts (corehq.apps.reports.tests.test_pillows_xforms.testReportXFormProcessing)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/reports/tests/test_pillows_xforms.py", line 151, in testConvertAndRestoreReportXFormDicts
    self.assertEqual(orig, restored)
AssertionError: {'doc_type': 'XFormInstance', 'domain': 'commtrack-public-demo', 'xmlns': 'http: [truncated]... != {'doc_type': 'XFormInstance', 'domain': 'commtrack-public-demo', 'xmlns': 'http: [truncated]...
Diff is 3091 characters long. Set self.maxDiff to None to see it.

----------------------------------------------------------------------
```

**Comment by benrudolph** [comment]
not sure how those tests ever worked/how this PR broke them, but this should fix it: https://github.com/dimagi/commcare-hq/pull/10960/commits/347fea06b97c5230d7d8f5aad1a746cac97bc25f

File: corehq/apps/reports/tests/test_pillows_xforms.py [hunk 4]
```diff
@@ -122,7 +122,7 @@
 
 
 
-class testReportXFormProcessing(TestCase):
+class TestReportXFormProcessing(SimpleTestCase):
```
Comment (dimagimon) [inline corehq/apps/reports/tests/test_pillows_xforms.py]: E303 too many blank lines (3)

**Comment by benrudolph** [comment]
this actually fixes the tests: https://github.com/dimagi/commcare-hq/pull/10960/commits/e0ba782cdbf061d8903ff988d0823b1bdb1d7fe4. and with the added bonus of making it a SimpleTestCase! 🏇

**Comment by benrudolph** [comment]
✅ tests are passing

**Comment by czue** [comment]
:+1: to merge from my side, but not going to push the button in case there's any index checking or coordination that needs to happen prior to deploy

**Comment by benrudolph** [comment]
nope i think we're all set

**Comment by benrudolph** [comment]
👬

**Pull request merged**